### ISSUE TITLE
There is a bug in the ordered list syntax check.

### ISSUE THREAD

**Issue by AbangSD** [DavidAnson/vscode-markdownlint#30]
For an example, there would be a warning, but nothing worng.
``` markdown
1. aaa

2. bbb

bbb

3. ccc

ccc

4. ddd

ddd
```

`[markdownlint] MD029/ol-prefix: Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]`

**Comment by DavidAnson** [comment]
The example above is actually three lists using numbering continuation to make it look like one list. You can see that by switching to "1." for all of the list items. [Example](https://markdown-it.github.io/#md3=%7B%22source%22%3A%221.%20aaa%5Cn%5Cn1.%20bbb%5Cn%5Cnbbb%5Cn%5Cn1.%20ccc%5Cn%5Cnccc%5Cn%5Cn1.%20ddd%5Cn%5Cnddd%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D)

`MD029` by default allows lists numbered as 1/2/3 or 1/1/1. Because the second and third lists in the example do not start with 1, the warning message is correct.

What is probably intended is to indent the list item continuations three spaces each to make this a single list with no warnings. [Example](https://markdown-it.github.io/#md3=%7B%22source%22%3A%221.%20aaa%5Cn%5Cn1.%20bbb%5Cn%5Cn%20%20%20bbb%5Cn%5Cn1.%20ccc%5Cn%5Cn%20%20%20ccc%5Cn%5Cn1.%20ddd%5Cn%5Cn%20%20%20ddd%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D)

**Comment by AbangSD** [comment]
I get it now, thank you a lot.
